src/main/scala/com/fulcrumgenomics/vcf/HapCutToVcf.scala

```diff
@@ -117,6 +123,32 @@ class HapCutToVcf
     vcfWriter.close()
   }
 
+  /** If fixing ambiguous (IUPAC) bases is desired, returns a new path to a VCF, otherwise returns the original input VCF. */
+  private def inputVcf: FilePath = if (!fixAmbiguousReferenceAlleles) this.vcf else {
+
+    val inputLines = PathUtil.extensionOf(this.vcf) match {
+      case Some(".vcf") => Io.readLines(this.vcf)
+      case Some(".vcf.gz")  =>
+        val stream = Io.toInputStream(this.vcf)
+        Source.fromInputStream(stream).withClose(() => stream.close()).getLines()
+      case Some(".bcf") => throw new IllegalArgumentException(s"BCF not supported when fixing ambiguity codes in: ${this.vcf}")
+      case _            => throw new IllegalArgumentException(s"Could not determine file type from ${this.vcf}")
+    }
+    val outputLines = inputLines.map { line =>
```

Comment:
Super lame that you have to do this and write to a file and can't filter on the fly and have VCFFileReader read from a  stream :/

Reply:
:/